src/internals/rules.rs

```diff
@@ -88,8 +100,8 @@ where
         })
 }
 
-impl<'a> From<(&'a yara_sys::YR_SCAN_CONTEXT, &'a yara_sys::YR_RULE)> for Rule<'a> {
-    fn from((context, rule): (&'a yara_sys::YR_SCAN_CONTEXT, &'a yara_sys::YR_RULE)) -> Self {
+impl<'a> From<&'a yara_sys::YR_RULE> for Rule<'a> {
+    fn from(rule: &'a yara_sys::YR_RULE) -> Self {
```

Comment:
Can you make this return a `RulesetRule` instead? So the `impl From<...> for Rule` depends on RulesetRule and not the opposite.

Reply:
For this, the signature would need to be changed to
```rust
 fn from(rule: &'a mut yara_sys::YR_RULE)
```
because the `inner` pointer of the `RulesetRule` needs to be mutable. Doing that, however, would also require changes in the ` MetadataIterator::from(rule)` and ` TagIterator::from(rule)` since they expect the rule to be immutable.

Moreover, when changing those Iterators too, this leads to a compiler error later:

```
error[E0382]: use of moved value: `rule`
   --> src/internals/rules.rs:112:38
    |
104 |     fn from(rule: &'a mut yara_sys::YR_RULE) -> Self {
    |             ---- move occurs because `rule` has type `&mut YR_RULE`, which does not implement the `Copy` trait
...
111 |         let metadatas = MetadataIterator::from(rule).map(Metadata::from).collect();
    |                                                ---- value moved here
112 |         let tags = TagIterator::from(rule)
    |                                      ^^^^ value used here after move
```